### PR TITLE
[TT-5590] add API ID if not provided while creating an API

### PR DESCRIPTION
**Pull request by jeffy-mathew**
## Description
added: generate API ID if user doesn't provide one while creating API.
fixed: creating API without APIID overwriting already existing API without APIID.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [ ] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
  - [ ] If new config option added, ensure that it can be set via ENV variable
- [ ] I have updated the documentation accordingly.
- [x] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] When updating library version must provide reason/explanation for this update.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] Check your code additions will not fail linting checks:
  - [x] `go fmt -s`
  - [x] `go vet`

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4101/merge
Commit: ce640e53a879c403eb18694081f5c23a984b1c4e 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2428015226)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4101/merge
Commit:  
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2428000792)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4101/merge
Commit: 0e358a790401c32dad0bdd45bc645ad39276cd6e 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2429333827)

File: apidef/api_definitions.go [hunk 1]
```diff
@@ -1047,6 +1050,10 @@ func (s *StringRegexMap) Init() error {
 	return nil
 }
 
+func (a *APIDefinition) GenerateAPIID() {
+	a.APIID = strings.Replace(uuid.NewV4().String(), "-", "", -1)
```
Comment (titpetric) [inline apidef/api_definitions.go:1054]: There's also a .ReplaceAll method there

**Review by titpetric — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: c0363aa48da89dabda3695db7901c148355da8f5 add API ID if not provided while creating an API (#4101)

[changelog]
added: generate API ID if user doesn't provide one while creating API.
fixed: creating API without APIID overwriting already existing API without APIID.
Triggered by: push (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2429926738)

**Pull request merged**